packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -54,10 +60,7 @@ const CommerceApiProvider = (props: CommerceApiProviderProps): ReactElement => {
         throwOnBadResponse: true
     }
 
-    // TODO: Initialize the api clients immediately, without waiting for an access token.
-    // See template-retail-react-app/app/commerce-api/index.js for inspiration
-    // especially how Proxy class can be used to wait for the access token and inject it to each request header.
-    const apiClients: ApiClients = {
+    const {current: apiClients} = useRef<ApiClients>({
```

Comment:
What's the reasoning for calling `useRef` on the api clients and the auth object?

What happens when the props for `CommerceApiProvider` get changed? This `useRef` value will not react to the props change. Initially, we wanted the api clients to be re-initialized whenever the props change.

Reply:
You are right Vincent. This should be `useState`.